src/utils.rs

```diff
@@ -159,6 +159,27 @@ macro_rules! impl_enum_decodable {
     };
 }
 
+// Same as try!, but for Option
+#[macro_export]
+macro_rules! try_opt {
+    ($expr:expr) => (match $expr {
+        Some(val) => val,
+        None => { return None; }
+    })
+}
+
+// Unwraps an iterator of Options. If any element is None, return None
+#[macro_export]
+macro_rules! try_opt_collect {
+    ($expr:expr) => ({
+        let mut res = Vec::with_capacity($expr.size_hint().0);
```

Comment:
Such a thing is already in the std libaries, by the way. See the first implementor of: http://doc.rust-lang.org/std/iter/trait.FromIterator.html.

```rust
try_opt_collect!(option_iter)
```
should be equivalent to
```rust
try_opt!(option_iter.collect::<Vec<_>>())
```